module4/lessons/Week2/BuildAnAPI.md

```diff
@@ -0,0 +1,403 @@
+            optionsBuilder.UseInMemoryDatabase("TestDatabase");
+
+            var context = new GoodBooksApiContext(optionsBuilder.Options);
+            context.Database.EnsureDeleted();
+            context.Database.EnsureCreated();
+
+            return context;
+        }
+    }
+}
+```
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### 2. Implement the Endpoint
+
+```c#
+using GoodBooksAPI.DataAccess;
+using GoodBooksAPI.Models;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GoodBooksAPI.Controllers
+{
+    // This will be the default route - controller gets replaced with the name of the controller.
+    // "/api/books"
+    [Route("/api/[controller]")]
+    [ApiController]
+    public class BooksController : ControllerBase
+    {
+        private GoodBooksApiContext _context;
+
+        public BooksController(GoodBooksApiContext context)
+        {
+            _context = context;
+        }
+
+        // GET "/api/books"
+        [HttpGet]
+        public ActionResult GetBooks()
+        {
+            List<Book> books = _context.Books.ToList();
+
+            // We could manually set the response code like this:
+            //Response.StatusCode = 200;
+
+            // API endpoints should return JSON, we are creating a new JSON result with our list of books.
+            return new JsonResult(books);
+        }
+    }
+}
+
+```
+
+</section>
+
+<section class='answer' markdown='1'>
+
+### 3. Experiment with Postman
+
+Now that our endpoing is tested and created, let's see it in action with Postman.
+
+1. Open your GoodBooks database in pgAdmin, and use the following script to insert some records into your development database
+```sql
+INSERT INTO books (title, description)
+VALUES
+    ('The Enigma of Eternity', 'A young physicist embarks on a perilous journey to unravel the mysteries of eternal existence in a world where time stands still.'),
+    ('Echoes of the Forgotten', 'In a post-apocalyptic world, a group of survivors must confront their inner demons and forge alliances to protect what remains of humanity'),
+    ('The Art of Serendipity', 'Two strangers with contrasting lives discover the transformative power of embracing the unexpected in a bustling city known for its chaos and unpredictability.');
+```
+2. Start your application, and make note of the localhost port number when chrome opens
+![local host port number](/assets/images/module4/week2/localhost_portnumber.png)
+3. Open Postman
+4. Send a GET request to the books route
+![Postman GET books](/assets/images/module4/week2/postman_getbooks.png)
+
+</section>
+
+### GET /books/1
+
+<section class='call-to-action' markdown='1'>
+**With a partner: Implement an API Endpoint that satisfies these specifications**  
+
+<section class='note' markdown='1'>
+* Path/Method: GET /api/books/1
+* Status Code: 200
+* Content: JSON containing the book with the id '1' (or whatever id is sent in through the path)
+</section>
```

Comment:
The formatting appears off in this section for some reason. The bullets do not render correctly.

Reply:
This is how it looks for me - are you seeing something different?

<img width="566" alt="Screenshot 2023-07-17 103143" src="https://github.com/turingschool/launch-curriculum/assets/32604200/681be668-d229-4c90-b519-b68b7c86c52f">